lxd/instance_instance_types.go

```diff
@@ -30,40 +32,59 @@ type instanceType struct {
 
 var instanceTypes map[string]map[string]*instanceType
 
-func instanceSaveCache() error {
-	if instanceTypes == nil {
-		return nil
+func instanceLoadFromDir(dir string) (map[string]map[string]*instanceType, error) {
+	newInstanceType := map[string]map[string]*instanceType{}
+	if !shared.PathExists(dir) || !shared.IsDir(dir) {
+		return newInstanceType, nil
 	}
 
-	data, err := yaml.Marshal(&instanceTypes)
+	entries, err := os.ReadDir(dir)
 	if err != nil {
-		return err
+		return nil, err
 	}
+	sort.Slice(entries, func(i, j int) bool {
```

Comment:
Newline between each stanza please

Reply:
Why are you sorting entries? This should have a comment if its important, so the knowledge is retained. Thanks